sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -1575,10 +1629,15 @@ private Connection getConnection() throws SQLException {
     public void processElement(ProcessContext context) throws Exception {
       // Only acquire the connection if we need to perform a read.
       Connection connection = getConnection();
+
       // PostgreSQL requires autocommit to be disabled to enable cursor streaming
       // see https://jdbc.postgresql.org/documentation/head/query.html#query-with-cursor
-      LOG.info("Autocommit has been disabled");
-      connection.setAutoCommit(false);
+      // This option is configurable as Informix will error
+      // if calling setAutoCommit on a non-logged database
+      if (disableAutoCommit) {
+        LOG.info("Autocommit has been disabled");
+        connection.setAutoCommit(false);
```

Comment:
It's not recommended to mutate connection here. Connection is cached once created.

For #31111 one can set "withDataSourceConfiguration" or "withDataSourceProviderFn" which creates DataSource that disabled autocommit, no need additional parameters here

Reply:
While I agree with your statement, this setAutoCommit statement has already been present in the code for some time. This PR is only making it conditional. Removing it entirely would bring a breaking change.

Reply:
I see, I understand now, it is this existing "setAutoCommit" call causing issues for certain jdbc provider. Yeah the existing code is not idea